### DISCUSSION TITLE
LaTeX development on separate branch

### DISCUSSION THREAD

**Discussion by IuryPiva** [bokeh/bokeh#11720] [General]
I propose that the remaining LaTeX work is done on a separate branch rather `branch 3-0`. When all the work is done, the whole branch can be merged to stable branch after squashing. The PR structure would look something like this:

![image](https://user-images.githubusercontent.com/26139868/136239393-3ba053be-5a2f-4b1b-b300-37e0fd3bf31f.png)

This solves the problems of:
 - reduce the number of baseline image changes
 - parallel PRs depending on other PRs
 - pull request containing changes of other PRs

The idea is to have the feature PRs only containing the code changes.

All updates to baselines files will be done only once on the LaTeX main branch when it's ready to be merged into `branch 3-0`.

**Comment by mattpap** [comment]
This is a valid development approach under given circumstances, however not applicable here. All your individual PRs have to be reviewed anyway. I would rather you focused on quality of individual changes, which would reduce my review burden and, in result, improve the turnaround time. I will definitely not review and merge one big PR. In fact I would reject it solely on principle.

**Comment by ianthomas23** [comment]
Perhaps I can rephrase the idea for extra clarity. The underlying problem is the requirement to provide both
1. PRs that are a single functional unit so that they are easy to review
2. Minimise the number of changes to visual test baseline images to keep the git repo size as small as possible

Providing one or the other is easy, but both at the same time is more difficult. The proposal here aims to provide both.

The expectation is that there will still be the same number of functional PRs, but they will be against the `mathtext-main-branch` rather than `branch-3.0`. They still need to be individually reviewed before acceptance on to the branch. But by keeping all the work on the separate branch there is an extra PR at the end to merge the branch to `branch-3.0`, and this will squash the commits from the functional PRs so that there is only one set of changes to the baseline images.

Of course the final merge of `mathtext-main-branch` to `branch-3.0` will lose the commit history, as is required to avoid multiple baseline image changes, but we can keep the branch hanging around for a while in case there are any bugs discovered in it.

**Comment by bryevdv** [comment]
I just want to reiterate that my understanding is that this is not about reviewing one big PR. Subject to individual changesets still being reviewable in isolation (always a requirement), I actually think different people work differently, and we should step out of the way and allow for contributors adopt whatever process best fits their individual circumstances or preferences with the least friction for them.

**Reply by bryevdv** [reply]
All that said, I do think getting the baselines out of the repo history (e.g into git LFS or similar) would also be a big improvement. But that change is not something I would be able to push forward myself.
